[RPD-2868] change the favorite status in the detailView

ehdudgnsTW opened the pull request:
<!--
  제목은 `[지라이슈번호] 작업 내용 요약`으로 작성해주세요.
-->

## 설명
DetailView에서 즐겨찾기의 상태를 변경할 수 있게 만들었습니다.

### 작업 내용

tableview에서 cell을 선택하였을때 상세페이지로 넘어갑니다.
그 상세 페이지는 셀의 데이터를 적용하였습니다.
상세페이지 안에서 즐겨찾기 버튼을 눌렀을시에 즐겨찾기 상태가 변경이 됩니다.

### 스크린샷

<img width="473" alt="스크린샷 2022-04-20 오전 11 17 33" src="https://user-images.githubusercontent.com/102941654/164133719-50723cbe-2c52-4dab-883c-452320b1515f.png">
<img width="473" alt="스크린샷 2022-04-20 오전 11 17 37" src="https://user-images.githubusercontent.com/102941654/164133757-0d698d5a-2781-4971-9407-e542d5bf745e.png">
<img width="473" alt="스크린샷 2022-04-20 오전 11 17 40" src="https://user-images.githubusercontent.com/102941654/164133825-9e3e5963-9c8a-4e12-9de2-c01cb912dfa9.png">
<img width="473" alt="스크린샷 2022-04-20 오전 11 17 43" src="https://user-images.githubusercontent.com/102941654/164133890-2095c4e6-2cc9-4012-befc-bd010abf36db.png">

## 기타

DetailViewReactor 파일이 추가 되었습니다.
MainViewReactor파일에서 init으로 데이터를 받아오는 부분을 rx로 받아오게 변경하였습니다.
Repository에서 csv파일 읽는 부분과 전체데이터를 읽어오는 부분을 합쳤습니다.

## 참고 링크
<!--
  (Optional) 검토시 참고할 만한 자료를 공유해주세요.
  - 기획 문서, 디자인 문서, slack 메시지의 링크를 추가합니다.
  - 개발시 참고했던 기술 문서, article 등을 추가합니다.
-->

ehdudgnsTW committed:
Rename isFavoriet to isFavorite

FavoriteDataModel에 있는 변수 명이 오타가 나서 변수명을 변경 하였습니다.

RPD-2868

ehdudgnsTW committed:
Update repository functions

init을 하기 위해서 만들었던 getCSVFileInit부분을 getTotalData()부분에서 파일의 유무에 따라서 실행시키도록 변경하였습니다.

RPD-2868

ehdudgnsTW committed:
Make UITableViewDelegate rxswift

delegate에서 셀을 선택해서 넘어가는 부분을 이제 bind를 통해서 셀을 선택했을때 현재 셀의 데이터를 가지고 넘어갈 수 있게 만들었습니다.

RPD-2868

ehdudgnsTW committed:
Remove UISearchResultUpdating extension

UISearchResultUpdating에서 하는 작업이 사라졌으므로 삭제를 하였습니다.

RPD-2868

ehdudgnsTW committed:
Fix LocationDataTableViewCell layout

검색할때 셀의 layout이 깨지는 현상이 발견되서 layout깨지는 부분을 수정 하였습니다.

RPD-2868

ehdudgnsTW committed:
Make from viewDidLoad to viewWillAppear

기존에 viewDidLoad에서 갱신했던 작업을 이제는 화면이 나타날때 마다 갱신하도록 변경 하였습니다.

RPD-2868

ehdudgnsTW committed:
Add DetailViewReactor

DetailView에서 버튼을 눌렀을때 즐겨찾기 상태가 변경되어야 하고 또한 로컬디비에 값도 변경이 되어야 해서 reactor를 만들었습니다.

RPD-2868

ehdudgnsTW committed:
Use DetailViewReactor in DetailViewController

DetailViewController에 새롭게 만들었던 DetailViewReactor를 연결 하였고 버튼 누르는 동작을 추가 하였습니다.

RPD-2868

ehdudgnsTW committed:
Fix DetailViewController init in MainViewController

DetailViewController를 init하는 부분에서 기존에는 model만 파라미터로 받았습니다.
reactor를 연결한뒤에 reactor도 설정을 해줘야 하므로 init부분의 파라미터를 추가하여서 reactor도 받을 수 있게 수정하였습니다.

RPD-2868

ehdudgnsTW committed:
Refactor MainViewReactor

기존에 init부분으로 전체 데이터를 받아왔었는데 detailview에서 변경 사항을 받아서 뷰가 나올때마다 전체 데이터를 갱신해야 할 필요가 있어 보였습니다.
그래서 뷰가 나올때마다 전체 데이터를 받아오도록 수정을 하였습니다.

RPD-2868

ehdudgnsTW committed:
Set cell selectioStyle

셀을 선택할때 selectionstyle을 변경 하였습니다.

RPD-2868

ehdudgnsTW committed:
Add json decode error case

json 파일을 읽어올때 실패할 경우에 대한 상황에 대해서 다시 json파일을 덮어씌우는 작업을 추가 하였습니다.

RPD-2868

hyeonsikTW commented:
도시 검색 후
![image](https://user-images.githubusercontent.com/97004324/164136388-406cb918-7d15-4e46-a77d-4c212c7d1312.png)
즐겨 찾기를 등록 한 후에 상세페이지에 들어간 뒤
![image](https://user-images.githubusercontent.com/97004324/164136469-655a18ce-78e6-4a9a-bec0-353f80cfb4d8.png)
다시 다른 도시의 즐겨찾기 버튼을 누르면 이전 검색 했던 결과가 나옵니다.
![image](https://user-images.githubusercontent.com/97004324/164137703-2949f136-f3e4-4e7a-9fd3-da57dc0194ef.png)

ehdudgnsTW committed:
Fix search issue where searchbar

서칭중에 상세페이지로 넘어갔다가 다시 검색할시에 화면이 갱신이 안되는 부분을 수정하였습니다.

RPD-2868

ehdudgnsTW commented:
> 도시 검색 후 ![image](https://user-images.githubusercontent.com/97004324/164136388-406cb918-7d15-4e46-a77d-4c212c7d1312.png) 즐겨 찾기를 등록 한 후에 상세페이지에 들어간 뒤 ![image](https://user-images.githubusercontent.com/97004324/164136469-655a18ce-78e6-4a9a-bec0-353f80cfb4d8.png) 다시 다른 도시의 즐겨찾기 버튼을 누르면 이전 검색 했던 결과가 나옵니다. ![image](https://user-images.githubusercontent.com/97004324/164137703-2949f136-f3e4-4e7a-9fd3-da57dc0194ef.png)

변경하였습니다

ehdudgnsTW committed:
Fix favoritebutton in locationcell action

검색중에 상세페이지로 갔다가 다시 돌아온뒤 검색어를 지우고 즐겨찾기 버튼을 눌렀을때 전에 검색어에 관련된 셀이 보이는 현상이 나와서 그부분을 수정하였습니다.

RPD-2868

nexmondTW commented:
수고하셨습니다.
